test_util/launch.py

```diff
@@ -154,58 +239,31 @@ def ssh_from_config(self, config):
-                private_key = load_string(config['ssh_info']['private_key_path'])
+                raise LauncherError('ValidationError', 'If a private_key_path is provided, '
+                                    'then the KeyName template parameter must be set')
+            private_key = load_string(config['ssh_info']['private_key_path'])
         if not key_name:
             key_name = config['stack_name']
-            generate_key = True
             private_key = self.boto_wrapper.create_key_pair(key_name)
+            temp_resources['key_name'] = key_name
             cf_params = config.get('parameters', list())
             cf_params.append({'ParameterKey': 'KeyName', 'ParameterValue': key_name})
             config['parameters'] = cf_params
-        return {
-            'delete_with_stack': generate_key,
-            'private_key': private_key,
-            'key_name': key_name,
-            'user': config.get('ssh_info', dict()).get('user', None)}
-
-    def test(self, info, test_cmd):
-        ssh_info = info['ssh']
-        for param in ['user', 'private_key']:
-            if param not in ssh_info:
-                raise LauncherError('MissingInfo', 'Cluster was created without providing ssh_info: {}'.format(param))
-        details = self.describe(info)
-        test_host = details['masters'][0]['public_ip']
-        # TODO: section should be generalized for use by other pr
-        with ssh.tunnel.tunnel(ssh_info['user'], ssh_info['private_key'], test_host) as test_tunnel:
-            return test_util.runner.integration_test(
-                tunnel=test_tunnel,
-                dcos_dns=test_host,
-                master_list=[m['private_ip'] for m in details['masters']],
-                agent_list=[a['private_ip'] for a in details['private_agents']],
-                public_agent_list=[a['private_ip'] for a in details['public_agents']],
-                aws_access_key_id=info['provider']['access_key_id'],
-                aws_secret_access_key=info['provider']['secret_access_key'],
-                region=info['provider']['region'],
-                test_cmd=test_cmd)
+        user = config.get('ssh_info', dict()).get('user', None)
+        ssh_info = {'private_key': private_key, 'user': user}
+        if user is None:
+            print('Testing not possible; user must be provided under ssh_info in config YAML')
+        if private_key is None:
+            print('Testing not possible; private_key_path must be provided under ssh_info in config YAML')
+        return ssh_info, temp_resources
 
     def get_stack(self, info):
-        """Returns the correct class interface depending how the AWS CF is configured
-        NOTE: only supports Simple Cloudformation currently
-        """
         try:
-            return DcosCfSimple(info['stack_name'], self.boto_wrapper)
+            return fetch_stack(info['stack_name'], self.boto_wrapper)
         except Exception as ex:
```

Comment:
We want to catch all and recast as a LauncherError here so that the error it self is readable on the command line